### PR TITLE
feat: Adds contract tests for Stream connection resource to CI

### PR DESCRIPTION
**Pull request by maastha**
## Proposed changes

_Jira ticket:_ https://jira.mongodb.org/browse/CLOUDP-230537

Adds contract tests for Stream connection resource to CI.

Link to any related issue(s): 

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as
  expected)
- [ ] This change requires a documentation update
- [ ] If changes include removal or addition of 3rd party GitHub actions, I updated our internal document. Reach out to the APIx Integration slack channel to get access to the internal document.

## Manual QA performed:

- [ ] cfn invoke for each of CRUDL/cfn test
- [ ] Updated resource in  [example](https://github.com/mongodb/mongodbatlas-cloudformation-resources/tree/master/examples)
- [ ] Published to AWS private registry
- [ ] Used the template in [example](https://github.com/mongodb/mongodbatlas-cloudformation-resources/tree/master/examples) to create and update a stack in AWS
- [ ] Deleted stack to ensure resources are deleted
- [ ] Created multiple resources in same stack
- [ ] Validated in Atlas UI
- [ ] Included screenshots

## Required Checklist:

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked that this change does not generate any credentials and that **they are NOT accidentally logged anywhere**.
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run `make fmt` and formatted my code
- [ ] For CFN Resources: I have released by changes in the private registry and proved by change
  works in Atlas

### CODE REVIEW

**Review by lantoli — APPROVED** [review]
this is great!

**Comment by lantoli** [comment]
jira ticket link missed in PR description

**Review by oarbusi — DISMISSED** [review]
LGTM

**Review by marcosuma — APPROVED** [review]

File: .github/workflows/contract-testing.yaml [hunk 1]
```diff
@@ -701,3 +704,44 @@ jobs:
...
+      - name: Run the Contract test
+        shell: bash
+        env:
+          MONGODB_ATLAS_PUBLIC_API_KEY: ${{ secrets.CLOUD_DEV_PUBLIC_KEY }}
+          MONGODB_ATLAS_PRIVATE_API_KEY: ${{ secrets.CLOUD_DEV_PRIVATE_KEY }}
+          MONGODB_ATLAS_ORG_ID: ${{ secrets.CLOUD_DEV_ORG_ID }}
+          MONGODB_ATLAS_OPS_MANAGER_URL: ${{ vars.MONGODB_ATLAS_BASE_URL }}
+          MONGODB_ATLAS_PROFILE: cfn-cloud-dev-github-action
+        run: |
+          pushd cfn-resources/stream-connection
```
Comment (lantoli) [inline .github/workflows/contract-testing.yaml:738]: similar to comment in Search Deployment, I'd use only cd if we'not using popd later

File: cfn-resources/stream-connection/Makefile [hunk 2]
```diff
@@ -17,3 +17,17 @@ debug:
...
+
+delete-test-resources:
+	@echo "==> Delete test resources used for contract testing"
+	./test/cfn-test-delete-inputs.sh
+
+run-contract-testing:
+	@echo "==> Run contract testing"
+	make build
+	sam local start-lambda &
+	cfn test --function-name TestEntrypoint --verbose
```
Comment (lantoli) [inline cfn-resources/stream-connection/Makefile:33]: have last line in the file with new line

**Review by lantoli — DISMISSED** [review]

File: cfn-resources/stream-connection/Makefile (hunk 2 above)
Comment (maastha) [reply cfn-resources/stream-connection/Makefile]: updated

File: .github/workflows/contract-testing.yaml (hunk 1 above)
Comment (maastha) [reply .github/workflows/contract-testing.yaml]: updated

**Review by oarbusi — APPROVED** [review]

**Pull request merged**